### PR TITLE
fix : Recording starts even when device not connected.

### PR DESCRIPTION
**Pull request by apoorva-raj**
**Fixes** #2093 

**Changes**: 
- Created a method in the PSLabSensor.java abstract class to check whether the android device has the required sensor or not.
- If the phone doesn't have the required sensor in-built and PS Lab is not connected, data recording won't start.

**Checklist**: 
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding any value.
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`.
- [x] I have reformatted code and fixed indentation in every file included in this pull request
- [x] My code does not contain any extra lines or extra spaces.
- [x] I have requested reviews from maintainers.

### CODE REVIEW

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
1. Always keep your branch updated
2. Reformat code
3. Fix issues reported by Codacy

**Comment by apoorva-raj** [comment]
Fixed the problems.

File: app/src/main/java/io/pslab/activity/MainActivity.java [hunk 1]
```diff
@@ -52,55 +53,76 @@
...
-
 public class MainActivity extends AppCompatActivity {
 
     private static final String TAG = "MainActivity";
-
+    private static final String TAG_DEVICE = "device";
+    private static final String TAG_INSTRUMENTS = "instruments";
+    private static final String TAG_ABOUTUS = "aboutUs";
+    private static final String TAG_PINLAYOUT = "pinLayout";
+    private static final String TAG_FAQ = "faq";
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/MainActivity.java:63]: There are lot of changes where they are just reshuffling content like this block. Please revert all of them back.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Don't edit (reshuffle) rest of the unrelated code.

**Comment by CloudyPadmal** [comment]
Also there are several commits related to gradle fixes. Update the PR and squash commits into one. https://github.com/fossasia/pslab-android#how-to-git-squash

**Comment by apoorva-raj** [comment]
@CloudyPadmal  I tried to change the settings in my IDE and then reformat the code but I don't know why it is still shuffling some parts. The app is working fine though and the fixes I made are reflected there.

File: app/src/main/java/io/pslab/models/PSLabSensor.java [hunk 2]
```diff
@@ -316,9 +318,12 @@ public boolean onOptionsItemSelected(MenuItem item) {
               will fire up. If user declines to give permission, don't do anything.
              */
             case R.id.record_data:
-                if (!isRecording) {
+                if (!isRecording && (!sensorNotFound() || scienceLab.isConnected())) {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/PSLabSensor.java:321]: According to this logic, if PSLab is connected, it doesn't check the sensor test right?

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/pslab/models/PSLabSensor.java (hunk 2 above)
Comment (apoorva-raj) [reply app/src/main/java/io/pslab/models/PSLabSensor.java]: It checks but user would be able to start the recording nevertheless since PSLab is connected.

**Review by CloudyPadmal — APPROVED** [review]

**Pull request merged**